applications/GeoMechanicsApplication/custom_elements/line_2d_5.hpp

```diff
@@ -0,0 +1,1314 @@
+            IntegrationMethod ThisMethod) const override
+        {
+            KRATOS_ERROR << "Jacobian is not square" << std::endl;
+        }
+
+        // ========================================================================================
+        // ========================================================================================
+        void ShapeFunctionsIntegrationPointsGradients(ShapeFunctionsGradientsType& rResult,
+            Vector& rDeterminantsOfJacobian, IntegrationMethod ThisMethod) const override
+        {
+            KRATOS_ERROR << "Jacobian is not square" << std::endl;
+        }
+
+        // ========================================================================================
+        // ========================================================================================
+        ///@}
+        ///@name Input and output
+        ///@{
+        /** Turn back information as a string.
+        @return String contains information about this geometry.
+        @see PrintData()
+        @see PrintInfo()
+        */
+        std::string Info() const override
+        {
+            return "1 dimensional line with 5 nodes in 2D space";
+        }
+
+        // ========================================================================================
+        // ========================================================================================
+        /** Print information about this object.
+         * @param rOStream Stream to print into it.
+         * @see PrintData()
+         * @see Info()
+         */
+        void PrintInfo(std::ostream& rOStream) const override
+        {
+            rOStream << "1 dimensional line with 5 nodes in 2D space";
+        }
+
+        // ========================================================================================
+        // ========================================================================================
+        /** Print geometry's data into given stream. Prints it's points 
+         * by the order they stored in the geometry and then center point of geometry.
+         * @param rOStream Stream to print into it.
+         * @see PrintInfo()
+         * @see Info()
+         */
+        void PrintData(std::ostream& rOStream) const override
+        {
+            BaseType::PrintData(rOStream);
+            std::cout << std::endl;
+            Matrix jacobian;
+            Jacobian(jacobian, PointType());
+            rOStream << "    Jacobian\t : " << jacobian;
+        }
+
+        // ========================================================================================
+        // ========================================================================================
+        /** Calculates the local gradients for all integration points for given integration method
+         */
+        virtual ShapeFunctionsGradientsType ShapeFunctionsLocalGradients(IntegrationMethod ThisMethod)
```

Comment:
It seems that a copy is made here. Shouldn't be "const IntegrationMethod &ThisMethod" instead of "IntegrationMethod ThisMethod"?

Reply:
Done